onnxruntime/contrib_ops/cpu/attnlstm/multi_scale_deformable_attention_avx512.cc

```diff
@@ -0,0 +1,489 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "contrib_ops/cpu/attnlstm/multi_scale_deformable_attention.h"
+
+#include <cstring>
+
+#include <immintrin.h>
+
+#define PREFETCH_ADDR_8x(ADDRESSES, HINT)   \
```

Comment:
I do some experiments and did not observe any performance improvement with prefetch:
```
Prefetch
8,1536,2048,2000,500,1870.861,58.767,262.167,255.941,1284.204,24.25,11.047,18.12,91.433,103.864,139.54,211.142,156.748,149.111,178.595,167.404,10.419,10.702
8,1536,2048,2000,500,1807.746,56.057,248.014,251.219,1242.228,22.828,10.773,17.882,86.668,98.41,133.67,204.428,149.403,145.92,173.402,164.619,10.024,9.731
8,1536,2048,2000,500,1794.646,51.394,246.093,250.193,1231.894,23.006,10.587,18.085,86.291,99.387,132.343,198.868,146.325,146.565,172.954,164.491,10.295,9.65
```
```
No prefetch
8,1536,2048,2000,500,1799.753,55.4,247.056,251.768,1234.378,22.707,10.54,18.269,85.853,99.765,132.763,201.615,146.181,146.129,172.483,164.52,10.119,9.568
8,1536,2048,2000,500,1811.867,51.295,247.524,251.866,1243.729,22.479,11.279,18.561,82.838,99.368,134.568,201.904,148.065,148.461,175.419,164.66,9.97,9.443
8,1536,2048,2000,500,1812.019,63.826,252.962,252.051,1239.884,22.842,10.632,18.304,86.22,98.757,134.22,200.818,147.775,147.681,174.57,164.301,10.007,9.796
```
Have you noticed any performance gains with prefetch? If not, are you considering removing it or altering the prefetch distance?

Reply:
I thought about it. Maybe we should remove the prefetching if it is not bringing us benefits. The reason is, my current implementation is issuing prefetch instructions as soon as the address is ready. It will be very hard to increase the prefetch distance because we do not have the address for the future cycles.